[BEE-27761] Updated based on editorial review comments (#74)
electric-cloud/EC-DslDeploy: help/help.adoc

|Artifact Version |The version of the artifact you want to deploy. If empty, the latest version is deployed.
NOTE: Use an empty version only in your `Dev` environment. In higher environments, it's recommended that you use a specific version that has been tested in lower environments.

|Pool or Resource |The name of the pool or resource where the artifact will be retrieved. The DSL is evaluated on the server so all system actions, such as file reads, will be evaluated on a server node.

Default value is `local`.
|Retrieve on All server nodes |Use this option when running in a cluster with a local workspace, as the artifact version needs to be retrieved on all nodes (as the system pick a random server node for each action). Use alongside a resource pool covering all server node agents for the parameter above.

Default is `false`
|Overwrite mode |Select to enable overwrite mode. If the object exists, all attributes and child objects are overwritten by the values in the DSL files.
|===

image:cloudbees-common-sda::cd-plugins/ec-dsldeploy/help/installdsl.png[installDsl parameters]

=== InstallDslFromDirectory

Deploy a DSL file structure from a directory on your {PRODUCT} server.

> This procedure main usage is to set up a CI process. It will extract your code from your SCM in a local directory on your `Dev` {PRODUCT} server, and deploy it from there.

Reviewer: ```suggestion
This procedure's main usage is to set up a CI process. It extracts your code from your SCM in a local directory on your `Dev` {PRODUCT} server, and deploys it.
```

[cols="1a,1a",options="header",]
|===
|Parameter |Description
|Directory Path |The location of the DSL code on your `Dev` {PRODUCT} server.
|Server Pool or Resource |The name of the pool or resource where the artifact will be retrieved. Use `.` if you have retrieved in your workspace.
|Overwrite mode |Select to enable overwrite mode. If the object exists, all attributes and child objects are overwritten by the values in the DSL files.

> |Additional argument(s)| For evalDsl calls, additional arguments such as  `--timeout` or `--debug`. Values used in `--timeout` will be used as timeouts for ectool, evalDsl, and the job step time out used to import the DSL.

Reviewer: ```suggestion
|Additional argument(s)| For evalDsl calls, additional arguments such as  `--timeout` or `--debug`. Values used in `--timeout` are used as timeouts for ectool, evalDsl, and the job step time out used to import the DSL.
```

|Ignore failures |Ignore failures during import of DSL files.
|Local mode |Do not send files to the server with help of clientFiles argument of the evalDsl operation in local mode.
|===

image:cloudbees-common-sda::cd-plugins/ec-dsldeploy/help/installdslfromdirectory.png[installDslFromDirectory parameters]

=== InstallProject

Deploy DSL file structure from a directory on your {PRODUCT} server as a project.

The main purpose of the procedure to be a part of installDslFromDirectory procedure.

[cols="1a,1a",options="header",]
|===
|Parameter |Description
|Project Name |Name of the project to install.
|Project directory |The location of the project DSL code on your `Dev` {PRODUCT} server.
|Overwrite mode |Select to enable overwrite mode. If the object exists, all attributes and child objects are overwritten by the values in the DSL files.

> |Additional argument(s)| For evalDsl calls, additional arguments such as  `--timeout` or `--debug`. Values used in `--timeout` will be used as timeouts for ectool, evalDsl, and the job step time out used to import the DSL.

Reviewer: ```suggestion
|Additional argument(s)| For evalDsl calls, additional arguments such as  `--timeout` or `--debug`. Values used in `--timeout` are used as timeouts for ectool, evalDsl, and the job step time out used to import the DSL.
```

|Suppress Defaults |Exclude from the generated DSL properties with default value.
|Suppress Parent |Exclude from the generated DSL properties referred to object parent.
|Include ACLs |Include in ACLs for objects.
|Include All Children |Include in the generated DSL all object children. If `True`, the value of the `Include Children` parameter is ignored.
|Include children |A comma-separated list of object children the DSL should be generated for.
|Include Children in Same File |Include in the generated DSL all object children. If `True`, the value of the `Include Children` parameter is ignored.
|Children in Different Files | A comma-separated list of patterns to include, like pipelines, procedures.`*`, applications.applicationTiers.components
|===

image:cloudbees-common-sda::cd-plugins/ec-dsldeploy/help/generatedsltodirectory.png[generateDslToDirectory parameters]

=== ImportDslFromGit

Import DSL file structure from a Git repository.

This procedure main usage is to set up a CI process.

[cols="1a,1a",options="header",]
|===
|Parameter |Description

> |Server Resource |The resource where the DSL files will be checked out from git and imported to the {PRODUCT} server.

Reviewer: ```suggestion
|Server Resource |The resource where the DSL files are checked out from Git and imported to the {PRODUCT} server.
```

> |Destination Directory |The directory on the resource where the source tree will be created and from where the DSL files are read to be imported in {PRODUCT} server.

Reviewer: ```suggestion
|Destination Directory |The directory on the resource where the source tree is created and from where the DSL files are read to be imported in {PRODUCT} server.
```

|Cleanup? |This option will delete the destination directory with the source tree after the job execution.
|Overwrite mode |Select to enable overwrite mode. If the object exists, all attributes and child objects are overwritten by the values in the DSL files.
|Configuration |The name of a saved SCM configuration.
|Git repository |The URL to the repository to pull from, for example `git://server/repo.git`.
|Commit Hash |The Commit Hash to update the index.

> NOTE: This will move the HEAD.

Reviewer: ```suggestion
NOTE: This moves the HEAD.
```
